joblib/parallel.py

```diff
@@ -939,6 +958,19 @@ def __call__(self, iterable):
         else:
             n_jobs = self._effective_n_jobs()
 
+        if (isinstance(self._backend, LokyBackend) and
+                self._reducer_extra is None):
+            # Create a context relative to this Parallel object. This context
+            # will be used when pickling batches to set the path of the folder
+            # where temporary resources created during this Parallel call will
+            # be located.  This is necessary to ensure that several Parallel
+            # object using the same resuable executor don't use the same
+            # temporary resources
+            self._reducer_extra = (
```

Comment:
Maybe call it `_reducer_context`?

Reply:
I though about this, but IMO `reducer_context != (context_id, temporary_folder_manager)`. I agree `reducer_extra` is not very specific though.

Reply:
Maybe rename it to `_batched_calls_reducer_callback` by making it a callable:

```python
def batched_calls_reducer_callback():
    self._backend._workers._temp_folder_manager.set_context(self._call_context_id)

self._batched_calls_reducer_callback = batched_calls_reducer_callback
```

Reply:
Good idea, let me know if the new version looks good.